File: Sources/LectureCopilot/ClassSession.swift
import Foundation

struct ClassSession: Codable {
    var id: UUID
    var title: String
    var startTime: Date
    var endTime: Date?
    var interactions: [ClassInteraction]
    var summary: String?
    var savePath: String?

    var duration: TimeInterval {
        (endTime ?? Date()).timeIntervalSince(startTime)
    }

    var recordedCount: Int {
        interactions.filter { !$0.answer.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty }.count
    }
}

struct ClassInteraction: Codable {
    var id: UUID
    var timestamp: Date
    var mode: String
    var screenshotRelativePath: String?
    var prompt: String
    var answer: String
}

struct ClassHUDSnapshot: Equatable {
    enum Phase: Equatable {
        case idle
        case running
        case summarizing
        case summaryReady
        case saved
    }

    var phase: Phase
    var elapsed: TimeInterval
    var interactionCount: Int
    var title: String
    var startClock: String

    static let idle = ClassHUDSnapshot(
        phase: .idle,
        elapsed: 0,
        interactionCount: 0,
        title: "Class Session",
        startClock: ""
    )

    static func durationText(_ interval: TimeInterval) -> String {
        let total = max(0, Int(interval))
        let hours = total / 3600
        let minutes = (total % 3600) / 60
        let seconds = total % 60
        if hours > 0 {
            return String(format: "%d:%02d:%02d", hours, minutes, seconds)
        }
        return String(format: "%02d:%02d", minutes, seconds)
    }

    static func clockText(_ date: Date) -> String {
        let formatter = DateFormatter()
        formatter.dateFormat = "h:mm a"
        return formatter.string(from: date)
    }
}

File: windows/LectureCopilot.Windows/Models/ClassSession.cs
namespace LectureCopilot.Windows.Models;

public sealed class ClassSession
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Title { get; set; } = "Class Session";
    public DateTimeOffset StartTime { get; set; } = DateTimeOffset.Now;
    public DateTimeOffset? EndTime { get; set; }
    public List<ClassInteraction> Interactions { get; set; } = [];
    public string? Summary { get; set; }
    public string? SavePath { get; set; }
    public TimeSpan Duration => (EndTime ?? DateTimeOffset.Now) - StartTime;
    public int RecordedCount => Interactions.Count(x => !string.IsNullOrWhiteSpace(x.Answer));
}

public sealed class ClassInteraction
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public DateTimeOffset Timestamp { get; set; } = DateTimeOffset.Now;
    public string Mode { get; set; } = "";
    public string? ScreenshotRelativePath { get; set; }
    public string Prompt { get; set; } = "";
    public string Answer { get; set; } = "";
}

public enum SessionPhase { Idle, Running, Ended, Summarizing, SummaryReady, Saved }

public sealed record ClassHudSnapshot(
    SessionPhase Phase,
    TimeSpan Elapsed,
    int InteractionCount,
    string Title,
    string StartClock)
{
    public static ClassHudSnapshot Idle { get; } = new(SessionPhase.Idle, TimeSpan.Zero, 0, "Class Session", "");

    public static string DurationText(TimeSpan duration) =>
        duration.TotalHours >= 1 ? duration.ToString(@"h\:mm\:ss") : duration.ToString(@"mm\:ss");
}
